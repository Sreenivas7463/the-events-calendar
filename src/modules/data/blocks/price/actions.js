/**
 * Internal dependencies
 */
import * as types from './types';
import * as utils from './utils';

export const setInitialState = ( props ) => ( {
	type: types.SET_INITIAL_STATE,
	payload: props,
} );

export const setCost = ( cost ) => ( {
	type: types.SET_PRICE_COST,
	payload: {
		cost,
	},
} );

export const setPosition = ( position ) => ( {
	type: types.SET_PRICE_POSITION,
	payload: {
		position,
	},
} );

export const togglePosition = ( showBefore ) => {
	return setPosition( utils.getPosition( showBefore ) );
};

export const setSymbol = ( symbol ) => ( {
	type: types.SET_PRICE_SYMBOL,
	payload: {
		symbol,
	},
} );
