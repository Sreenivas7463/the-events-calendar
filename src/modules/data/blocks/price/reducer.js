/**
 * Wordpress dependenciess
 */
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import { settings, priceSettings } from '@moderntribe/common/utils/globals';
import { string, globals } from '@moderntribe/common/utils';
import * as types from './types';

const position = string.isTruthy( settings() && settings().reverseCurrencyPosition )
	? 'suffix'
	: 'prefix';

export const getDefaultState = () => {
	if ( globals.wpCoreEditor.isCleanNewPost() ) {
		return;
	}

	const postId = globals.wpCoreEditor.getCurrentPostId();
	const entityRecord = globals.wpCore.getEntityRecord( 'postType', 'tribe_events', postId );

	DEFAULT_STATE = {
		position: entityRecord.meta._EventCurrencyPosition,
		symbol: entityRecord.meta._EventCurrencySymbol,
		cost: entityRecord.meta._EventCost,
		description: '',
	};
};

export let DEFAULT_STATE = {
	position: priceSettings() && priceSettings().defaultCurrencyPosition
		? priceSettings().defaultCurrencyPosition
		: position,
	symbol: priceSettings() && priceSettings().defaultCurrencySymbol
		? priceSettings().defaultCurrencySymbol
		: __( '$', 'the-events-calendar' ),
	cost: '',
	description: '',
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_PRICE_COST:
			return {
				...state,
				cost: action.payload.cost,
			};
		case types.SET_PRICE_POSITION:
			return {
				...state,
				position: action.payload.position,
			};
		case types.SET_PRICE_SYMBOL:
			return {
				...state,
				symbol: action.payload.symbol,
			};
		case types.SET_PRICE_DESCRIPTION:
			return {
				...state,
				description: action.payload.description,
			};
		default:
			return state;
	}
};
