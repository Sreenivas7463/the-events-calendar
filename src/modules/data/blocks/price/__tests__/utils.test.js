/**
 * Internal dependencies
 */
import * as utils from '@moderntribe/events/data/blocks/price/utils';

describe( '[STORE] - Price utils', () => {
	it( 'Should get position', () => {
		expect( utils.getPosition( true ) ).toMatchSnapshot();
		expect( utils.getPosition( false ) ).toMatchSnapshot();
	} );
} );
