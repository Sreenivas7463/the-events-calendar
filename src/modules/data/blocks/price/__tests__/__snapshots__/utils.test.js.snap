// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - Price utils Should get position 1`] = `"prefix"`;

exports[`[STORE] - Price utils Should get position 2`] = `"suffix"`;
